pyomo/contrib/mindtpy/mip_solve.py

```diff
@@ -45,13 +45,11 @@ def solve_OA_master(solve_data, config):
         solve_data.mip.component_data_objects(Objective, active=True))
     main_objective.deactivate()
 
-    sign_adjust = 1 if main_objective.sense == minimize else -1
-    if MindtPy.find_component('MindtPy_oa_obj') is not None:
-        del MindtPy.MindtPy_oa_obj
+    sign_adjust = 1 if main_objective.sense == minimize else - 1
+    MindtPy.del_component('MindtPy_oa_obj')
 
-    if config.add_slack == True:
-        if MindtPy.find_component('MindtPy_penalty_expr') is not None:
-            del MindtPy.MindtPy_penalty_expr
+    if config.add_slack is True:
```

Comment:
@qtothec: Unless the config can take something other than True/False (e.g., is 'None' an acceptable value?), I think you should use Python's build in bool context and not check that the value is exactly True/False.  That is:
```python
if config.add_slack:
```
The same comment applies to lines 73 and 90 below.